integ-test/src/test/java/org/opensearch/sql/jdbc/CursorIT.java

```diff
@@ -0,0 +1,198 @@
+    Statement stmt = connection.createStatement();
+
+    for (var table : List.of(TEST_INDEX_CALCS, TEST_INDEX_ONLINE, TEST_INDEX_BANK, TEST_INDEX_ACCOUNT)) {
+      var query = String.format("SELECT COUNT(*) FROM %s", table);
+      ResultSet rs = stmt.executeQuery(query);
+      int rows = 0;
+      for (; rs.next(); rows++) ;
+
+      var restResponse = executeRestQuery(query, null);
+      assertEquals(rows, restResponse.getInt("total"));
+    }
+    connection.close();
+  }
+
+  @Test
+  @SneakyThrows
+  public void select_all_small_table_big_cursor() {
+    connection = DriverManager.getConnection(getConnectionString());
+    Statement stmt = connection.createStatement();
+
+    for (var table : List.of(TEST_INDEX_CALCS, TEST_INDEX_BANK)) {
+      var query = String.format("SELECT COUNT(*) FROM %s", table);
+      stmt.setFetchSize(200);
+      ResultSet rs = stmt.executeQuery(query);
+      int rows = 0;
+      for (; rs.next(); rows++) ;
+
+      var restResponse = executeRestQuery(query, null);
+      assertEquals(rows, restResponse.getInt("total"));
+    }
+    connection.close();
+  }
+
+  @Test
+  @SneakyThrows
+  public void select_all_small_table_small_cursor() {
+    connection = DriverManager.getConnection(getConnectionString());
+    Statement stmt = connection.createStatement();
+
+    for (var table : List.of(TEST_INDEX_CALCS, TEST_INDEX_BANK)) {
+      var query = String.format("SELECT * FROM %s", table);
+      stmt.setFetchSize(3);
+      ResultSet rs = stmt.executeQuery(query);
+      int rows = 0;
+      for (; rs.next(); rows++) ;
+
+      var restResponse = executeRestQuery(query, null);
+      assertEquals(rows, restResponse.getInt("total"));
+    }
+    connection.close();
+  }
+
+  @Test
+  @SneakyThrows
+  public void select_all_big_table_small_cursor() {
+    connection = DriverManager.getConnection(getConnectionString());
+    Statement stmt = connection.createStatement();
+
+    for (var table : List.of(TEST_INDEX_ONLINE, TEST_INDEX_ACCOUNT)) {
+      var query = String.format("SELECT * FROM %s", table);
+      stmt.setFetchSize(10);
+      ResultSet rs = stmt.executeQuery(query);
+      int rows = 0;
+      for (; rs.next(); rows++) ;
+
+      var restResponse = executeRestQuery(query, null);
+      assertEquals(rows, restResponse.getInt("total"));
+    }
+    connection.close();
+  }
+
+  @Test
+  @SneakyThrows
+  public void select_all_big_table_big_cursor() {
+    connection = DriverManager.getConnection(getConnectionString());
+    Statement stmt = connection.createStatement();
+
+    for (var table : List.of(TEST_INDEX_ONLINE, TEST_INDEX_ACCOUNT)) {
+      var query = String.format("SELECT * FROM %s", table);
+      stmt.setFetchSize(500);
+      ResultSet rs = stmt.executeQuery(query);
+      int rows = 0;
+      for (; rs.next(); rows++) ;
+
+      var restResponse = executeRestQuery(query, null);
+      assertEquals(rows, restResponse.getInt("total"));
+    }
+    connection.close();
```

Comment:
Why does connection have to close after each test when the connection string doesn't seem to change?

Reply:
Good point. Thank you, fixed